components/core/tests/search_test_utils.cpp

```
@@ -0,0 +1,107 @@
+auto make_logtype_dict(vector<vector<variant<string_view, char>>> const& entries)
+        -> MockLogTypeDictionary {
+    MockLogTypeDictionary dict;
+    logtype_dictionary_id_t id{0};
+    for (auto const& entry : entries) {
+        dict.add_entry(generate_expected_logtype_string(entry), id++);
+    }
+    return dict;
+}
+
+auto generate_expected_logtype_string(vector<variant<string_view, char>> const& tokens) -> string {
+    string result;
```

Cppcheck (2.19.0)

[style] The function 'get_num_placeholders' is never used.

(unusedFunction)